### DISCUSSION TITLE
Bevy Game Jam (after v0.6 release)

### DISCUSSION THREAD

**Discussion by Supermath101** [bevyengine/bevy#2364] [General]
Please discuss details here: https://hackmd.io/@Supermath101/BevyGameJam.

**Comment by NiklasEi** [comment]
I love the idea, thanks for bringing it up again!
A few suggestions:

1. I would like to reuse existing infrastructure for this kind of event and feel like running the whole thing on [itch.io](https://itch.io) is our best option.
2. About rules:
    * Use Bevy
    * General rules could come from our [Code of conduct](https://github.com/bevyengine/bevy/blob/main/CODE_OF_CONDUCT.md)
3. I would prefer judging by all participants over a panel of judges. This can be done very nicely on itch.
4. Use Bevy Discord for mentoring, organizing, team finding and so on
    * Get more people into the community
    * Reuse existing help channels

**Reply by NiklasEi** [reply]
Following [discussion on discord](https://discord.com/channels/691052431525675048/810632276701347920/855895876224679977)
5. Require the code to be public (no license requirements)

**Comment by alice-i-cecile** [comment]
Other notes:
- week long to allow time to iron out issues (and accommodate different schedules)
- just use GMT, start on a weekend one week after 0.6 launches
- arbitrarily pick a theme
- borrow criteria from other jams

**Reply by RustyStriker** [reply]
I think 1 week after 0.6 launches can be a bit too short(tho i dont know how much breaking changes are there), yet people need to both update, make sure they know/understand/remember how to use the engine and ofc to make sure no weird and nasty bug will show up that during the event itself.

**Reply by NathanSWard** [reply]
Yea that's true.
Especially if out goal is to encourage `0.6` usage we need to allow people some time to get used to the `0.6` API changes :)

**Reply by Nilirad** [reply]
We also need to wait for plugin maintainers to update for 0.6.

**Comment by NathanSWard** [comment]
> Any competition/prizes? If so, paid for by whom?

I'm more than happy to be a primary contributor for prizes (if that's something we want to do). 
I can also help with the organizational aspects if desired 😊

**Reply by NathanSWard** [reply]
I'm curious (this is more a question for @cart) but would you be open sponsorships for this?
(E.g. reaching out to companies and seeing if theyll sponsor a game jam?)

**Reply by cart** [reply]
Totally!

**Reply by mockersf** [reply]
Embark? Heroic Labs?

**Reply by NathanSWard** [reply]
If anyone has any specific groups we should reach out to post them here. 
I'll begin to draft a letter/email we can send out to possible sponsors and see if they'll support us :smile:

**Reply by alice-i-cecile** [reply]
Nvidia and AMD and Alienware and so on might bite; especially with the new renderer in this release.

**Comment by alice-i-cecile** [comment]
@Supermath101 could you prepare a little spreadsheet (or table) of evaluation criteria from other game jams?

GMTK and Ludum Dare come to mind as big ones to emulate.

**Reply by Supermath101** [reply]
I'm a little busy the rest of this weekend, but I'm free most of next week. I'll try though.

**Comment by Byteron** [comment]
I'd like to point at the Godot Wild Jam, it is the best Jam I have been part of.

When:
every second Friday of the month. So it is a monthly jam.

Duration: 
9 Days. This encloses 2 weekends, to be friendly towards working people.
(given bevy is not yet feature complete, it generally takes longer to do things with it, so a longer duration would be appropriate anyways, I think.)

Theme: 
Voted. There is a period before the jam where users make suggestions and vote for them.
There is a second phase where a theme is picked at random or from a moderator from the top some voted theme suggestions, so it's not obvious which theme it will be before it begins

Wildcards:
Every Jam, there are so-called wild cards. They are kind of optional extra challenges. Some examples from past jams:
- add adaptive music to your game
- make a mechanic that incorporates real time (as in 24 hour day real time)
- include a home / hub world in your game
- add carrots to your game (quite a few of them are contextual to what's happening in the community...)
- add at least one accessibility setting in your game
and so on. Those are pretty cool, they motivate to try certain things that you might not have thought of, or encourage to concern yourself with concepts you know nothing about. After all, game jams are, for me anyways, a place of experimentation and learning. (and fun! :D)
There are usually three of those every jam.

here is the jam page of the currently running godot wild jam for more details.
https://itch.io/jam/godot-wild-jam-34

**Reply by NathanSWard** [reply]
- They 9 day duration I actually really like (especially for us who have day jobs)

- I think for this first game jam, it may just be easier to have the poll locally on discord and then for any other game jams we can extend the pool to the greater community.  (OHHH, actually we could say something like "If you want to help vote on the theme for the game jam, come join the discord :D")

- Wild-cards sounds like a really fun idea! Though I'm curious if we can do something that's a spin-off of these to avoid _stealing_ the Godot Game Jam's wildcards (unless of course, more game jams do something similar to this?)

**Reply by greaka** [reply]
Wild cards are not a specific thing at all. The game jams I have participated in so far all used "Diversifiers" as the term for extra challenges/constrains.

**Reply by BorisBoutillier** [reply]
I also think that the 9 day duration is really good.
I don't think doing it monthly is yet a think for bevy, but doing everytime after a new bevy release will be a way to give developper a reason to come back and check what has been added to the engine, and increase visibility of the engine, with a regular 'hype' ,  without 'burning out'

**Reply by fluffysquirrels** [reply]
Re: duration, the game jam I'm currently doing starts on a Friday and ends after a week and 3 days on the following Sunday. This gives an extra Friday evening to 9-to-5 folks compared to the Godot approach. I like both but prefer the extra Friday :)

**Comment by Supermath101** [comment]
As discussed [here](https://www.reddit.com/r/rust_gamedev/comments/o4597k/bevy_game_jam_after_v06_release_very_early_in_the/h2fqmjy?utm_source=share&utm_medium=web2x&context=3), maybe this should be for all Rust game engines in general, not just Bevy. The commenter made a point that Bevy might not be popular enough to warrant a Game Jam just for it, specifically if this is the first Rust Game Jam ever. I suggested there should be a few different competition tracks, one for each major game engine, one for all lesser known game engines, and one for no game engine at all. Any thoughts?

**Reply by alice-i-cecile** [reply]
Honestly, I think it's popular enough. It also levels the playing field and makes coordination much simpler.

We only need a dozen or so participants to make it a success.

**Reply by NathanSWard** [reply]
Yep, and to be frank, if the game jam isn't just for Bevy specific projects, we lose a large aspect as to why we're doing this game jam in the first place (e.g. to get people to use bevy haha...)

**Reply by Nilirad** [reply]
10k GitHub stars is popular enough to my standards 😄

Also, a well done "marketing campaign" will attract lots of people. I imagine it like that:

- 0.6 release: at the end of the devlog, mention in the last section that, say, 3 weeks ahead, there will be the game jam.
- As the date approaches, make another post reminding the game jam and disclosing some more details, like the rules.

**Reply by cart** [reply]
Yeah I think this should be the first of many "officially sanctioned" Bevy game jams (for the reasons others have discussed above). I think a generic Rust game jam is a great idea, but we have specific goals for this one. (I also left some comments in the reddit thread linked above).

**Comment by mirkoRainer** [comment]
Random internet person interested here. Perhaps a poll for a game theme? Then have a random selection on start day?

**Reply by alice-i-cecile** [reply]
Yep, that or "Cart arbitrarily decides" are the most popular options so far :)

**Reply by mirkoRainer** [reply]
Plot twist. Theme is "cart". 

Mario cart clones for everyone!

**Comment by VitalyAnkh** [comment]
I wonder when will v0.6 be released? I can’t wait to see it! 🥰

**Reply by alice-i-cecile** [reply]
A month or two seems about right :) Plan is to wrap up the rendering rework (see https://github.com/bevyengine/bevy/discussions/2351 for some progress), merge the PR backlog and then ship.

**Reply by Barugon** [reply]
If you don't mind me asking, what's the holdup?

**Reply by alice-i-cecile** [reply]
My unofficial opinion is that there's lots of small details to get right in rendering, and there was a lot of migration work to do. This is made worse by the fact that a good chunk of the active contributors don't have a rendering background (like me), and so we can't help out as directly as with other features.

Steps remaining:
- fully remove the old renderer from main
- quash any deal-breaking new bugs
- prepare release notes
- (optional) implement pipelining
- (optional) merge PRs that have been sitting in ready-for-final-review

**Comment by Nilirad** [comment]
I have an idea for a new evaluation criteria: usage of new features. If a contendant makes good use of the most important features introduced by the latest engine update, they will receive a higher score in that category.

**Reply by mirkoRainer** [reply]
I think this would imply too much restriction. I'm ignorant of what change would be released for 0.6, but would it make more sense to pick a theme that would lend itself to using the new features instead of awarding just for use. 

The goal is to drum up Bevy interest so I would think the lowest barrier to entry and success will gain the most participation. 

However, there definitely could be a winner category for best use of new features or something.

**Reply by NathanSWard** [reply]
I'm personally not the biggest fan of this. This would definitly give an advantage to anyone who has already used the engine before and shun away new users. 

Also, (afaik) the biggest change is the rendering backend, which unless they're writing low level rendering code, won't affect the average user.

**Reply by Nilirad** [reply]
I agree with both of you. I think that at the end usage of new features will come up naturally as they simplify processes that usually are done manually, especially on the ECS side.

**Reply by TheNeikos** [reply]
I think that if a new feature is useful in games, it will naturally enhance them so would already be part of the evaluation criteria.

**Comment by jcornaz** [comment]
Now that bevy 0.6 is out of the door, and that the migration of plugins is [in progress](https://lib.rs/crates/bevy/rev), maybe we can start concretise the game jam? Like picking the dates and creating the entry on itch.io?

I'm eager to jam with bevy :-)

**Reply by alice-i-cecile** [reply]
@jcornaz and @Supermath101 are you two comfortable leading this? I'd be happy to provide support :)

**Reply by mockersf** [reply]
We talked about doing the jam on Discord, the date that would fit most people is from the 2022/02/11 to the 2022/02/21, to not be too close to other jams, and have a week plus the two weekends around.

Waiting for a few more confirmations and I'll set it up on itch.io if no one objects

**Reply by alice-i-cecile** [reply]
Excellent, that works for me. Hopefully I can coordinate that with an initial launch of the new book, to both help beginners and do a big marketing push.

**Reply by jcornaz** [reply]
@mockersf that dates look good to me :+1: 

@alice-i-cecile, I'd be glad to participate. Though, there are probably others interested too. Maybe, could we open a `game-jam` category of channel on the bevy discord? I think theses two channels would be a good start: 

* `theme-suggestion`: Suggest a theme for the next game jam
* `general`: Discuss about the jam event. It may eventually be split into multiple channels when we feel the need.

**Reply by alice-i-cecile** [reply]
~I'll start a pair of threads on Discord!~ We're convening in the 0.6 Game Jam thread in #community.

**Comment by mirkoRainer** [comment]
Theming the jam around The Bevy Book would be 🔥
And would likely have marketing synergy.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
